### PR TITLE
RES-574: Requirements doc

### PR DESCRIPTION
**Pull request by rhyolight**

### CODE REVIEW

File: REQUIREMENTS.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+Highbrow will be used to provide graphical animations of HTM systems for use within educational materials to accompany papers and videos that explain HTM technology.
+
+## Visualization Requirements
+
+Highbrow should display an animated view of an HTM system as it runs. The HTM state data could be read in from a batched file or streamed into Highbrow directly from a running HTM system.
+
+## Network Configuration
+
+An HTM Network must be defined by an configuration. This YAML file should allow the complete configuration of a network, it's architecture, and all its components. This configuration allows the complete rendering of the network, sans neuron state. Neuron state updates will be make using a data stream (defined below).
```
Comment (lscheinkman) [inline REQUIREMENTS.md]: Is it required for the configuration to be written in `YAML`?

File: REQUIREMENTS.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+Because data is particular to specific network configurations, it might be necessary to "type check" data before attempting to stream. Data files might be able to properly identify the network configurations they work with in some type of header information.
+
+## Playback Requirements
+
+In order to incorporate Highbrow content into video materials, scenarios must be set up where Highbrow visualizations can be run and exported into a standard movie format, preferably 1920x1080 at 30fps. Exports should either have a transparent or solid green background.
+
+## Controls
+
+Users must be able to move around and through the animation as it runs. Fly controls seem to be the most flexible, using the typical `WASD` for strafing and `←↑→↓` to rotate the camera. Holding `shift` should increase the speed of movement and rotation.
+
```
Comment (lscheinkman) [inline REQUIREMENTS.md]: Do you need it to record and replay the animation control movements?

File: REQUIREMENTS.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+## Renderable Objects
+
+Highbrow should provide a way to visualize an HTM network in three dimensional space, including the following key renderable objects:
+
+- Neuron
+- Layer
+- Mini Column <small>[1]</small>
+- Cortical Column
+- Network
+
```
Comment (lscheinkman) [inline REQUIREMENTS.md]: What about charts and other data?

File: REQUIREMENTS.md (hunk 2 above)
Comment (rhyolight) [inline REQUIREMENTS.md]: I don't have a requirement for that.

File: REQUIREMENTS.md (hunk 3 above)
Comment (rhyolight) [inline REQUIREMENTS.md]: No requirements for charting or rendering input data, predictions, anomalies, etc. Just the cellular states.

File: REQUIREMENTS.md (hunk 1 above)
Comment (rhyolight) [inline REQUIREMENTS.md]: No, will clarify.

**Comment by rhyolight** [comment]
@lscheinkman Any other questions or concerns?

**Comment by lscheinkman** [comment]
👍  LGTM

**Pull request merged**